superbench/common/utils/topo_aware.py

```diff
@@ -35,37 +38,41 @@ def search(self, pattern, string, flags=0):
         return self.matched
 
 
-def gen_ibstat_file(ibstat_file):
-    """Generate ibstat file for each node with specified path.
+def gen_ibstat_file(host_list, ibstat_file):
+    """Generate ibstat file in each node with specified path.
 
     Args:
+        host_list (list): list of VM read from hostfile.
         ibstat_file (str): path of ibstat output.
     """
-    from mpi4py import MPI
-
-    if not MPI.Is_initialized():
-        MPI.Init()
-
-    comm = MPI.COMM_WORLD
-    name = MPI.Get_processor_name()
-
-    # The command to fetch ibstat info
-    cmd = r"ibstat | grep -Po 'System image GUID: \K\S+$'"
-    output = os.popen(cmd)
-    ibstat = 'VM_hostname ' + name + '\n' + str(output.read())
-
-    # Fetch all ibstate from each node
-    ibstats = comm.allgather(ibstat)
-
-    ibstate_file_path = Path(ibstat_file)
-
-    # Filter the duplicate info
-    ibstat_infos = set(ibstats)
-
-    with ibstate_file_path.open(mode='w') as f:
-        for ibstat_info in ibstat_infos:
-            f.write(ibstat_info)
-    MPI.Finalize()
+    try:
+        # Parse ssh config
+        config = SSHConfig.from_path('/root/.ssh/config')
+
+        # Append each host config into host_config
+        host_config = []
+        for hostx in host_list:
+            hostx_config = config.lookup(hostx)
+            port = hostx_config['port']
+            private_key = hostx_config['identityfile'][0]
+            host_config.append(HostConfig(port=int(port), private_key=private_key))
+        # Fetch ibstat info from each host
+        client = ParallelSSHClient(host_list, host_config=host_config)
+        cmd = r"ibstat | grep -Po 'System image GUID: \K\S+$'"
```

Comment:
`ibstat` won't work when accelerated networking is enabled when creating VM, pls use ` cat /sys/class/infiniband/*/sys_image_guid | tr -d :` instead

Reply:
```sh
 parallel-ssh -i -t 5 -p 512 -h hostfile 'cat /sys/class/infiniband/*/sys_image_guid | tr -d :' | grep '[[:alnum:]]\{16\}' | sort | uniq
```
you can run it with parallel-ssh to get all guid

Reply:
Thanks for the suggestion!
>  parallel-ssh -i -t 5 -p 512 -h hostfile 'cat /sys/class/infiniband/*/sys_image_guid | tr -d :' | grep '[[:alnum:]]\{16\}' | sort | uniq
But if my understanding is correct, this command may cause two issues:
1. All guid with random order which is hard to map the guid with hostname
2. those guid missing the correct prefix can not be used to generate config as the less info to map phy host to guid in topo-aware

Reply:
udpated, please have a check.

Reply:
Did you find the solution for the above two issues or they are not the issue any more? 
I believe you just removed prefix `0x`when mapping for #2, but not quite sure #1.

Reply:
Yea, your are right for #2. 
For #1, as the sequential relationship is implied in the output format from pssh is like:
```
[1] 03:03:00 [SUCCESS] <hostname#1>
guid#1
guid#2
guid#3
[2] 03:03:00 [SUCCESS] <hostname#2>
guid#1
guid#2
guid#3
```
so we can just replace the `[1] 03:03:00 [SUCCESS]` to get the correct info